WebApi/WebApi/Library/LibraryManager.cs

```diff
@@ -0,0 +1,237 @@
+                }
+            }
+
+            return currentBook;
+        }
+
+        /// <summary>
+        /// Get author by id
+        /// </summary>
+        /// <param name="id">author's id</param>
+        /// <returns>current author</returns>
+        public Author GetAuthor(int id)
+        {
+            Author currentAuthor = null;
+            foreach (var author in this.authors)
+            {
+                if (author.Id == id)
+                {
+                    currentAuthor = author;
+                }
+                else
+                {
+                    throw new ArgumentException("No author with this id");
+                }
+            }
+
+            return currentAuthor;
+        }
+
+        /// <summary>
+        /// Change book's values
+        /// </summary>
+        /// <param name="id">book's id</param>
+        /// <param name="newBook">current book</param>
+        public void UpdateBook(int id, Book newBook)
+        {
+            Book currentBook = null;
+            foreach (var book in this.books)
+            {
+                if (book.Id == id)
+                {
+                    currentBook = book;
+                }
+                else
+                {
+                    throw new ArgumentException("No book with this id");
+                }
+            }
+
+            if (currentBook != null)
+            {
+                currentBook.Name = newBook.Name;
+                currentBook.Year = newBook.Year;
+                currentBook.AuthorId = newBook.AuthorId;
+            }
+        }
+
+        /// <summary>
+        /// Change author's values
+        /// </summary>
+        /// <param name="id">author's id</param>
+        /// <param name="newAuthor">current author</param>
+        public void UpdateAuthor(int id, Author newAuthor)
+        {
+            Author currentAuthor = null;
+            foreach (var author in this.authors)
+            {
+                if (author.Id == id)
+                {
+                    currentAuthor = author;
+                }
+                else
+                {
+                    throw new ArgumentException("No book with this id");
+                }
+            }
+
+            if (currentAuthor != null)
+            {
+                currentAuthor.Name = newAuthor.Name;
+                currentAuthor.BirthYear = newAuthor.BirthYear;
+            }
+        }
+
+        /// <summary>
+        /// Remove book
+        /// </summary>
+        /// <param name="id">book's id</param>
+        public void DeleteBook(int id)
+        {
+            Book currentBook = null;
+            foreach (var book in this.books)
+            {
+                if (book.Id == id)
+                {
+                    currentBook = book;
+                }
+                else
+                {
+                    throw new ArgumentException("No book with this id");
```

Comment:
And here... Your code will work only in the scenario when the first item in the list is same as you are requesting for